starknet_py/net/client_models.py

```diff
@@ -239,10 +244,9 @@ class TransactionReceipt:
     events: List[Event] = field(default_factory=list)
     l2_to_l1_messages: List[L2toL1Message] = field(default_factory=list)
 
-    execution_status: Optional[
-        TransactionExecutionStatus
-    ] = None  # gateway/pending receipt field
+    execution_status: Optional[TransactionExecutionStatus] = None
     finality_status: Optional[TransactionFinalityStatus] = None
+    # TODO (#1179): remove status field once devnet is updated
```

Comment:
Since you are already making changes to the receipt and removing gateway fields, maybe we could make execution and finality statuses non-optional as per the spec, or is there something blocking this?

Reply:
That was the plan at the beginning, but since we are still using the old devnet with RPC v0.3.0 it would require some additional work to make those changes now. @Radinyn is already working on updating devnet and once it is done we will do a proper refactoring that will include changes that you mentioned. Apologies for not making this clear in this PR.